linenoise.c

```diff
@@ -1168,7 +1170,11 @@ int linenoiseHistorySave(const char *filename) {
     fp = fopen(filename,"w");
     umask(old_umask);
     if (fp == NULL) return -1;
-    chmod(filename,S_IRUSR|S_IWUSR);
+    if (chmod(filename,S_IRUSR|S_IWUSR) < 0) {
```

Comment:
so FYI, all of this is here because of an old bug in redis where the history file (passwords et. al.) were being saved world readable if the user did not have a sane umask set.

So the fix, if one is needed, should be in the context of solving the actual problem more correctly - not just quieting Coverity.

Personally I would just cast the return on chmod() (void).